service/matching/handler.go

```diff
@@ -40,16 +40,18 @@ var _ m.TChanMatchingService = (*Handler)(nil)
 type Handler struct {
 	taskPersistence persistence.TaskManager
 	engine          Engine
+	config          *Config
 	metricsClient   metrics.Client
 	startWG         sync.WaitGroup
 	service.Service
 }
 
 // NewHandler creates a thrift handler for the history service
-func NewHandler(taskPersistence persistence.TaskManager, sVice service.Service) (*Handler, []thrift.TChanServer) {
+func NewHandler(taskPersistence persistence.TaskManager, sVice service.Service, config *Config) (*Handler, []thrift.TChanServer) {
```

Comment:
The "convention" I followed is to have config after the service.Service object. So how about this convention:
NewHandler(service.Service, config, extra objects needed)